plugin.video.redbull.tv/resources/lib/redbulltv_client.py

```diff
@@ -0,0 +1,110 @@
+import re, urllib2
+from lib import utils
+
+class RedbullTVClient(object):
+    REDBULL_API = "https://appletv-v2.redbull.tv/"
+    ROOT_MENU = [
+        {"title": "Discover", "url": REDBULL_API + "views/discover", "is_content":False},
+        {"title": "TV", "url": REDBULL_API + "views/tv", "is_content":False},
+        {"title": "Channels", "url": REDBULL_API + "views/channels", "is_content":False},
+        {"title": "Calendar", "url": REDBULL_API + "views/calendar", "is_content":False},
+        {"title": "Search", "url": REDBULL_API + "search?q=", "is_content":False},
+    ]
+
+    def __init__(self, resolution=None):
+        self.resolution = resolution
+
+    @staticmethod
+    def get_resolution_code(video_resolution_id):
+        return {
+            "0" : "320x180",
+            "1" : "426x240",
+            "2" : "640x360",
+            "3" : "960x540",
+            "4" : "1280x720",
+        }.get(video_resolution_id, "1920x1080")
+
+    def get_stream_details(self, element):
+        name = element.findtext("title")
+        description = element.findtext("description")
+        image = element.find("image").get("src1080")
+        url = element.findtext("mediaURL")
+
+        # Try find the specific stream based on the users preferences
+        try:
+            playlists = urllib2.urlopen(url).read()
+            resolution = self.get_resolution_code(self.resolution)
+            media_url = re.search(
+                "RESOLUTION=" + resolution + ".*\n(.*)",
+                playlists).group(1)
+        except Exception:
+            print "Couldn't find " + resolution + " stream", "Reverting to default quality"
```

Comment:
Please use `xbmc.log` with a minimum level of LOGDEBUG, thanks FYI for exceptions a level of LOGERROR is advisable.
https://codedocs.xyz/xbmc/xbmc/group__python__xbmc.html#ga3cd3eff2195727144ad6fbabd4744961